test/makeCompleteAction.test.js

```diff
@@ -161,4 +161,84 @@ describe("makeCompleteAction", () => {
       });
     });
   });
+
+  describe("with customSuccessPayload option", () => {
+    const customSuccessPayload = payload => payload;
+
+    it("should return a request action with a custom payload", () => {
+      const action = makeCompleteAction("Products", {
+        customSuccessPayload
+      });
+
+      expect(action("request", { test: 123 })).toEqual({
+        type: "GET_PRODUCTS",
+        test: 123
+      });
+    });
+
+    it("should return a success action with a custom payload", () => {
+      const action = makeCompleteAction("Products", {
+        customSuccessPayload,
+        payload: { test: 123 }
```

Comment:
I don't quite get it why you're using this `payload` object here, since you're already passing the `customSuccessPayload`. Also, you don't have to test this `payload` specifically here because we already have some tests for it.

Reply:
I think this was confusing because my code wasn't actually using the payload functions, now they are. The functions are passed what we pass into this payload variable. The result takes the values from inside the payload and assigns onto the action directly.

Reply:
Hmm... Right, I see that in your tests you're doing this:

```js
it("should return a success action with a custom payload", () => {
        const action = makeCompleteAction("Products", {
          customSuccessPayload: customPayload,
          payload: { test: 123 }
        });

        expect(action("success", { test: 123 })).toEqual({
          type: "GET_PRODUCTS_SUCCESSFUL",
          test: 123
        });
      });
```

But why not do this?

```js
it("should return a success action with a custom payload", () => {
        const action = makeCompleteAction("Products", {
          customSuccessPayload: customPayload
          // no default payload here!
        });

        expect(action("success", { test: 123 })).toEqual({
          type: "GET_PRODUCTS_SUCCESSFUL",
          test: 123
        });
      });
```

Reply:
Oh, I see now!

Just one thing, I'm seeing here that in the test: `"should return a success action with a custom payload"`, you have a default payload: `payload: { test: 123 }` and you're passing a custom payload in the "action": `expect(action("success", { test: 123 }))...`.

I understand that your intention here is to make very clear that we're using the "custom" payload and not the "default" one, right? But you're only using this in only two ("success", "failure") of the tests (this is missing in the "request" one). Do you mind to put this in the "request" test as well?

Reply:
Never mind, I understand your original point, and I just had some extra code in the success test that I didn't need. removed it, the tests still pass, and the test now reads cleaner. Cleaned up, sorry again.